tofu/geom/_core.py

```diff
@@ -5760,6 +5760,86 @@ def calc_min_rho_from_Plasma2D(self, plasma, t=None, log='min',
                                     for vv in np.split(val, lind, axis=-1)])
         return vals, pts, t
 
+    def get_inspector(self, ff):
+        out = inspect.signature(ff)
+        pars = out.parameters.values()
+        na = np.sum([(pp.kind == pp.POSITIONAL_OR_KEYWORD
+                      and pp.default is pp.empty) for pp in pars])
+        kw = [pp.name for pp in pars if (pp.kind == pp.POSITIONAL_OR_KEYWORD
+                                         and pp.default is not pp.empty)]
+        return na, kw
+
+    def check_ff(self, ff, t=None, ani=None):
+        time_steps = -1
+        # .. Checking basic definition of function ..........................
+        str_error = "Input emissivity function (ff): "
+        assert hasattr(ff, '__call__'), (str_error
```

Comment:
However, small caveat:  the sue of assert, though compact, is not the best option in my sense.
If statements would be bit better because:
* more readable code structure
* The error messages are defined only if the check is not passed (with assert, the message has to be defined even if there is no error).
* This [technical reason](https://softwareengineering.stackexchange.com/questions/225956/python-assert-vs-if-return)

I'll take over from here just to turn asserts into if, for good habits

Reply:
ok, the code was originally yours :)